.github/workflows/on-pr-merge.yml

```diff
@@ -50,14 +46,27 @@ jobs:
           python -m pip install --upgrade --upgrade-strategy eager -r requirements/ci_merge-requirements.txt -e ./requirements
           KADALU_VERSION=devel make cli-build
 
+      - name: Check to build and push base image or not
+        id: build_base
+        run: |
+          # same conditions as mentioned in submit workflow
+          FILES=$(git show --pretty="" --name-only)
+          build_base='no'
+          if grep -e 'CHANGELOG.md' <<< "$b" && [ $(wc -w <<< "$b") == 1 ]; then
+            build_base='yes'
+          elif grep -e 'extras/Dockerfile.builder' <<< "$FILES"; then
+            build_base='yes'
+          fi
+          echo ::set-output name=out::$build_base
       - name: Builder Image
+        if: "contains(steps.build_base.outputs.build_base, 'yes')"
         uses: docker/build-push-action@v2
         timeout-minutes: 120
         with:
           context: .
           file: extras/Dockerfile.builder
           platforms: linux/amd64
-          push: false
+          push: true
           load: true
           tags: |
             docker.io/kadalu/builder:latest
```

Comment:
If we are pushing this image, should this be 'devel' tag? or latest is fine?

latest we add for release only for now IMO.

Reply:
we need `latest` as in every component image we are referring the base image with `latest` tag